### PR TITLE
fix: eindstatus duplicate check used less precise zaak einddatum instead of eindstatus datumstatusgezet resulting in duplicate check false positives

### LINKED ISSUE #262: SoapUI testcase "Suites4SociaalDomein Natuurlijk Peroon" updateZaak sometimes fails
It is likely a timing issue that should never happen around changes in status. During the processing of statuses there is a check to make sure that it is not going to add a duplicate status by checking the datumstatusgezet on existing statuses and the datumstatusgezet on the current one its trying to add. If they are the same, it will throw an error.

The root cause is probably around the detection of changes of statuses. The timing issue is merely a symptom of something happening that shouldn't. The relevant adapter is called "SetResultaatAndStatus".

### PR DESCRIPTION
**Pull request by DelanoWAF**
Incorrect XSLT condition caused loss of datetime information. Resolved now by replacing the condition in question.

Resolves #262

### CODE REVIEW

File: src/main/configurations/Translate/Zgw/Zaken/Model/ZgwStatus.xslt [hunk 1]
```diff
@@ -21,9 +21,12 @@
 
     <xsl:variable name="DatumStatusGezetFromParams" as="xs:string">
         <xsl:choose>
-            <xsl:when test="$ZgwStatusType/ZgwStatusType/isEindstatus = 'true' and not($ZdsEinddatum = 'Undefined')">
+            <xsl:when test="$ZgwStatusType/ZgwStatusType/isEindstatus = 'true' and (string-length($ZdsEinddatum) > 0) and (string-length($ZdsStatusDatum) = 0)">
                 <xsl:value-of select="zgw:toZgwDatetime(concat($ZdsEinddatum,'000000'))"/>
             </xsl:when>
+            <xsl:when test="$ZgwStatusType/ZgwStatusType/isEindstatus = 'true'">
+                <xsl:value-of select="zgw:toZgwDatetime($ZdsStatusDatum)"/>
+            </xsl:when>
```
Comment (MLenterman) [inline src/main/configurations/Translate/Zgw/Zaken/Model/ZgwStatus.xslt:29]: This is doing the same thing as the "otherwise", so this one is not needed.

**Review by MLenterman — CHANGES_REQUESTED** [review]
Im missing the correction on the testcase.

**Comment by DelanoWAF** [comment]
Performing additional investigation into the safeguard that prevents duplicate results. It appears to be incorrect still and requires adjustment.

**Comment by DelanoWAF** [comment]
Delays in test cases are necessary. The code still checks the datetime of statusDatumGezet to ensure that no two statuses were set at the exact same time. However, the safeguard for status type has been added to prevent a status from being added to a case more than once. 

Original code did include a minor version of the safeguard but this appeared to not enough, as well as not functioning correctly.

**Review by MLenterman — APPROVED** [review]

**Review by MLenterman — APPROVED** [review]

**Pull request merged**